src/main/python/image_preprocessing/face_manager.py

```diff
@@ -96,6 +103,9 @@ def blur_face(self, image):
         # We make a deep copy of an image to avoid problems with shallow copies.
         image_copy = image.copy()
```

Comment:
I also don't think we should make a copy inside the function. A function should only work with the parameters provided to it, so if a copy is really necessary, the copy operation should happen before the method call. Do we actually get errors with shallow copies?